tests/test_driver.sh

```diff
@@ -0,0 +1,128 @@
+# vim: set ts=4 sw=4 tw=100 et:
+# POSIX sh test driver to reduce dependency on fish in tests
+
+# macOS has really weird default IFS behavior that splits output in random places, and the trailing
+# backspace is to prevent \n from being gobbled up by the subshell output substitution.
+# Folks, this is why you should use fish!
+IFS="$(printf "\n\b")"
+
+# The first argument is the path to the script to launch; all remaining arguments are forwarded to
+# the script.
+fish_script="$1"
+shift 1
+script_args="${@}"
+
+die() {
+    if test "$#" -ge 0; then
+        printf "%s\n" "$@" 1>&2
+    fi
+    exit 1
+}
+
+# To keep things sane and to make error messages comprehensible, do not use relative paths anywhere
+# in this script. Instead, make all paths relative to one of these or $homedir."
+TESTS_ROOT="$(dirname "$0")"
+BUILD_ROOT="${TESTS_ROOT}/.."
+FISH_ROOT="${BUILD_ROOT}/.."
```

Comment:
I think this is assuming that the build directory is directly inside the fish source directory, which is often not the case. You can reproduce by making a separate directory and running `cmake /path/to/fish`; the tests will fail.

Reply:
Do you know if I have to explicitly pass these as command line arguments or environment variables from CMake on invocation or if there's something already set I can use?

Reply:
nvm. We are (I am?) already passing it as a command line argument, just not using it in the sh script.